src/root_files/_common/404.html

```diff
@@ -97,11 +97,12 @@
             }
         };
     }
-
     var all_lang_regex = new RegExp('(' + all_languages.join('|') + ')', 'i');
     var is_from_404 = location.search &&
         location.search.toLowerCase().indexOf('from_404')  !== -1;
-    var redirectFn = routeHandler.isApp2() && !is_from_404 ? routeHandler.redirectToReactRoute : routeHandler.redirect;
+    var is_from_app2 = location.search &&
+        location.search.toLowerCase().indexOf('app2')  !== -1;
+    var redirectFn = routeHandler.isApp2() && !is_from_404 ? routeHandler.redirectToReactRoute : is_from_app2 ? routeHandler.redirectToReactRoute : routeHandler.redirect;
```

Comment:
please use brackets to indicate which statements go together

Reply:
actually, can't this be changed to:
```
var redirectFn = ((routeHandler.isApp2() && !is_from_404) || is_from_app2) ? routeHandler.redirectToReactRoute : routeHandler.redirect;
```

Reply:
Absolutely. Done.